jobsdb/jobsdb.go

```diff
@@ -395,15 +395,20 @@ HandleT is the main type implementing the database for implementing
 jobs. The caller must call the SetUp function on a HandleT object
 */
 type HandleT struct {
-	dbHandle                      *sql.DB
-	ownerType                     OwnerType
-	tablePrefix                   string
-	datasetList                   []dataSetT
-	datasetRangeList              []dataSetRangeT
-	dsListLock                    lock.DSListLocker
-	dsMigrationLock               sync.RWMutex
-	MinDSRetentionPeriod          time.Duration
-	MaxDSRetentionPeriod          time.Duration
+	dbHandle         *sql.DB
+	ownerType        OwnerType
+	tablePrefix      string
+	datasetList      []dataSetT
+	datasetRangeList []dataSetRangeT
+	dsListLock       lock.DSListLocker
+	dsMigrationLock  sync.RWMutex
+	// dsRetentionMap is protected by dsMigrationLock
+	dsRetentionMap       map[string]time.Time
```

Comment:
The reason I created it is to keep in-memory the real timestamp of when the table was created instead of relying on fields inside like `created_at` that could be susceptible to external changes and requires an additional query. I thought I'd propose it to simplify a bit, wdyt?

Reply:
jobsDB current usage patterns don't promote the use of this map:

1. jobsDB operates in 3 modes: reader, writer, read-writer
2. We might have multiple jobsDB instances for the same set of tables, e.g. 1 reader & 1 writer
3. Writers only create (new) tables
4. Reader both create (migrated) tables and delete tables

Based on the above we risk ending up with:

- jobsDBs with split brain, i.e. different entries in this map
- writer jobsDBs with a map of ever-increasing entries

Finally, the min & max retention isn't really concerned about when the table was created, it has to do with the actual data that the table holds, e.g. a migrated table may be created now, but it will hold past data.